pkg/controller/schedule_controller.go

```diff
@@ -18,188 +18,123 @@ package controller
-				c.logger.Info("Creating schedule ", scheduleName)
-				//Init Prometheus metrics to 0 to have them flowing up
-				metrics.InitSchedule(scheduleName)
-			},
-		},
-	)
-
-	return c
+func (c *scheduleReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	s := kube.NewPeriodicalEnqueueSource(c.logger, mgr.GetClient(), &velerov1.ScheduleList{}, scheduleSyncPeriod)
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&velerov1.Schedule{}).
+		Watches(s, nil).
+		Complete(c)
 }
 
-func (c *scheduleController) enqueueAllEnabledSchedules() {
-	schedules, err := c.schedulesLister.Schedules(c.namespace).List(labels.NewSelector())
-	if err != nil {
-		c.logger.WithError(errors.WithStack(err)).Error("Error listing Schedules")
-		return
-	}
+// +kubebuilder:rbac:groups=velero.io,resources=schedules,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=velero.io,resources=schedules/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=velero.io,resources=backups,verbs=create
 
-	for _, schedule := range schedules {
-		if schedule.Status.Phase != api.SchedulePhaseEnabled {
-			continue
-		}
+func (c *scheduleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := c.logger.WithField("schedule", req.String())
 
-		key, err := cache.MetaNamespaceKeyFunc(schedule)
-		if err != nil {
-			c.logger.WithError(errors.WithStack(err)).WithField("schedule", schedule).Error("Error creating queue key, item not added to queue")
-			continue
-		}
-		c.queue.Add(key)
+	log.Debug("Getting schedule")
+	schedule := &velerov1.Schedule{}
+	if err := c.Get(ctx, req.NamespacedName, schedule); err != nil {
+		c.logger.WithError(err).Error("error getting schedule")
+		return ctrl.Result{}, nil
 	}
-}
 
-func (c *scheduleController) processSchedule(key string) error {
-	log := c.logger.WithField("key", key)
-
-	log.Debug("Running processSchedule")
-	ns, name, err := cache.SplitMetaNamespaceKey(key)
-	if err != nil {
-		return errors.Wrap(err, "error splitting queue key")
+	if schedule.Status.Phase != "" &&
+		schedule.Status.Phase != velerov1.SchedulePhaseNew &&
+		schedule.Status.Phase != velerov1.SchedulePhaseEnabled {
+		c.logger.Debugf("the schedule phase isn't %s or %s, skip", velerov1.SchedulePhaseNew, velerov1.SchedulePhaseEnabled)
+		return ctrl.Result{}, nil
 	}
 
-	log.Debug("Getting Schedule")
-	schedule, err := c.schedulesLister.Schedules(ns).Get(name)
-	if err != nil {
-		// schedule no longer exists
-		if apierrors.IsNotFound(err) {
-			log.WithError(err).Debug("Schedule not found")
-			return nil
-		}
-		return errors.Wrap(err, "error getting Schedule")
-	}
+	c.metrics.InitSchedule(schedule.Name)
 
-	switch schedule.Status.Phase {
-	case "", api.SchedulePhaseNew, api.SchedulePhaseEnabled:
-		// valid phase for processing
-	default:
-		return nil
+	patchHelper, err := patch.NewHelper(schedule, c.Client)
+	if err != nil {
+		c.logger.WithError(err).Error("error new patch helper")
```

Comment:
it would be better to use `log` instead of `c.logger` as it has more context?

There are a few such cases in this func, please also make update.